devito/types/equation.py

```diff
@@ -157,6 +157,211 @@ def __str__(self):
+
+    Parameters
+    ----------
+    lhs : Function or SparseFunction
+        The left-hand side.
+    rhs : expr-like, optional
+        The right-hand side. Defaults to 0.
+    subdomain : SubDomain, optional
+        To restrict the evalaution of the relation to a particular sub-region in the
+        computational domain.
+
+    Examples
+    --------
+    Ge may be used to express a relation.
+
+    >>> class InnerDomain(SubDomain):
+            name = 'inner'
+
+            def define(self, dimensions):
+                return {d: ('middle', 2, 2) for d in dimensions}
+    >>> inner_domain = InnerDomain()
+    >>> grid = Grid(shape=(8, 8), subdomains=(inner_domain,))
+    >>> g = Function(name='g', shape=grid.shape, dimensions=grid.dimensions)
+    >>> x, y = grid.dimensions
+    >>> xi, yi = grid.subdomains['inner'].dimensions
+    >>> cond = Ge(g[xi, yi], rhs, subdomain=grid.subdomains['inner'])
+    >>> ci = ConditionalDimension(name='ci', parent=yi, condition=cond)
+    >>> f = Function(name='f', shape=grid.shape, dimensions=(xi, ci))
+    >>> Eq1 = Eq(f[x, y], g[x, y])
+    """
+
+    def __init__(self, lhs, rhs=0, subdomain=None, **kwargs):
+        obj = sympy.Ge.__new__(self, lhs, rhs, **kwargs)
+        obj._subdomain = subdomain
+        return obj
+
+    @property
+    def subdomain(self):
+        """The SubDomain in which the Ge is defined."""
+        return self._subdomain
+
+
+class Gt(sympy.Gt):
+    """
+    A strict-greater-than (">") relation between two objects, the left-hand side and the
+    right-hand side. Can be used to build conditionals but not directly to
+    construct an Operator.
+
+    The left-hand side may be a Function or a SparseFunction. The right-hand
+    side may be any arbitrary expressions with numbers, Dimensions, Constants,
+    Functions and SparseFunctions as operands.
+
+    Parameters
+    ----------
+    lhs : Function or SparseFunction
+        The left-hand side.
+    rhs : expr-like, optional
+        The right-hand side. Defaults to 0.
+    subdomain : SubDomain, optional
+        To restrict the evalaution of the relation to a particular sub-region in the
+        computational domain.
+
+    Examples
+    --------
+    Gt may be used to express a relation.
+
+    >>> class InnerDomain(SubDomain):
+            name = 'inner'
+
+            def define(self, dimensions):
+                return {d: ('middle', 2, 2) for d in dimensions}
+    >>> inner_domain = InnerDomain()
+    >>> grid = Grid(shape=(8, 8), subdomains=(inner_domain,))
+    >>> g = Function(name='g', shape=grid.shape, dimensions=grid.dimensions)
+    >>> x, y = grid.dimensions
+    >>> xi, yi = grid.subdomains['inner'].dimensions
+    >>> cond = Gt(g[xi, yi], rhs, subdomain=grid.subdomains['inner'])
+    >>> ci = ConditionalDimension(name='ci', parent=yi, condition=cond)
+    >>> f = Function(name='f', shape=grid.shape, dimensions=(xi, ci))
+    >>> Eq1 = Eq(f[x, y], g[x, y])
```

Comment:
I think some of these need to have the output in the docstring if it returns something when run

Reply:
though unfortunately the example isn't really computing anything...

Reply:
SHould I add the snippet of the generated code that contains loops+conditional?

Reply:
Is this fine now?